src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/DualNodesSendReceiveTest.java

```diff
@@ -165,9 +185,55 @@ void receiverCanView() {
   void senderCanView() {
     final EthClientStub firstNode = NodeUtils.client(firstOrionLauncher.clientPort(), firstHttpClient);
 
-    final String digest = sendTransaction(firstNode, PK_1_B_64, PK_2_B_64);
+    final String digest = sendTransaction(firstNode, PK_1_B_64, PK_3_B_64);
     final byte[] receivedPayload = viewTransaction(firstNode, PK_1_B_64, digest);
 
     assertTransaction(receivedPayload);
   }
+
+  @Test
+  void sendToMultipleRecipientsOnOneOrionUsesAllRecipientsKeysToEncryptButStripsAllOtherNodesKeys() {
+    final EthClientStub firstNode = NodeUtils.client(firstOrionLauncher.clientPort(), firstHttpClient);
+    final EthClientStub secondNode = NodeUtils.client(secondOrionLauncher.clientPort(), secondHttpClient);
+
+    final String digest = sendTransaction(firstNode, PK_1_B_64, PK_4_B_64, PK_5_B_64);
+    byte[] receivedPayload = viewTransaction(secondNode, PK_4_B_64, digest);
+    assertTransaction(receivedPayload);
+
+    receivedPayload = viewTransaction(secondNode, PK_5_B_64, digest);
+    assertTransaction(receivedPayload);
+
+    assertThatThrownBy(() -> viewTransaction(secondNode, PK_3_B_64, digest)).isInstanceOf(AssertionFailedError.class);
```

Comment:
This a bit hard to read with so many assertThatThrownBy, perhaps add an assertTransactionNotReceived or something like that to make this more readable

Reply:
nice